inst/private/python_ipc_pytave.m

```diff
@@ -59,17 +61,50 @@
 
   info.prelines = 0;
 
-  % TODO: bit wasteful to repeat this every call
-  headers = python_header();
-  pyexec(headers)
+  if isempty(have_headers)
+    pyexec(strjoin({'import sys',
+                    'import sympy',
+                    'import sympy as sp',
+                    'from sympy import __version__ as spver',
+                    'from sympy import *',
+                    'from sympy.logic.boolalg import Boolean, BooleanFunction',
+                    'from sympy.core.relational import Relational',
+                    '# temporary? for piecewise support',
+                    'from sympy.functions.elementary.piecewise import ExprCondPair',
+                    'from sympy.integrals.risch import NonElementaryIntegral',
+                    'from sympy.matrices.expressions.matexpr import MatrixElement',
+                    '# for hypergeometric',
+                    'from sympy.functions.special.hyper import TupleArg',
+                    'from sympy.utilities.iterables import uniq',
+                    'import copy',
+                    'import binascii',
+                    'import struct',
+                    'import codecs',
+                    'import xml.etree.ElementTree as ET',
+                    'from distutils.version import LooseVersion',
+                    'def dictdiff(a, b):',
+                    '    """ keys from a that are not in b, used by evalpy() """',
+                    '    n = dict()',
+                    '    for k in a:',
+                    '        if not k in b:',
+                    '            n[k] = a[k]',
+                    '    return n',
+                    'def Version(v):',
+                    '    # short but not quite right: https://github.com/cbm755/octsympy/pull/320',
+                    '    return LooseVersion(v.replace(".dev", ""))',
```

Comment:
can we remove the commas from the end of these lines?  some have some don't, I guess I prefer none.